packages/webapp/src/components/Animals/Inventory/index.tsx

```diff
@@ -14,44 +14,85 @@
+import styles from './styles.module.scss';
+import clsx from 'clsx';
+
+export type SearchProps = {
+  searchString: string | null | undefined;
+  setSearchString: Dispatch<SetStateAction<string[]>>;
+  placeHolderText: string;
+  searchResultsText: string;
+};
 
 const PureAnimalInventory = ({
-  tableData,
+  filteredInventory,
   animalsColumns,
-  theme,
-  isMobile,
+  zIndexBase,
+  backgroundColor,
+  isDesktop,
+  searchProps,
 }: {
-  tableData: AnimalInventory[];
+  filteredInventory: AnimalInventory[];
   animalsColumns: TableV2Column[];
-  theme: DefaultTheme;
-  isMobile: boolean;
+  zIndexBase: number;
+  backgroundColor: string;
+  isDesktop: boolean;
+  searchProps: SearchProps;
 }) => {
+  const { searchString, setSearchString, placeHolderText, searchResultsText } = searchProps;
+  const hasSearchResults = filteredInventory.length !== 0;
+
   return (
     <Layout
       classes={{
         container: {
-          backgroundColor: theme.palette.background.paper,
-          borderRadius: '8px',
-          border: '1px solid var(--Colors-Primary-Primary-teal-50)',
-          marginTop: '16px',
+          backgroundColor: backgroundColor,
+          borderRadius: isDesktop && '8px',
+          border: isDesktop && '1px solid var(--Colors-Primary-Primary-teal-50)',
+          marginTop: isDesktop && '16px',
+          padding: !isDesktop ? '0px' : '16px',
         },
       }}
       hasWhiteBackground
       footer={false}
     >
-      <Table
-        kind={TableKind.V2}
-        alternatingRowColor={true}
-        columns={animalsColumns}
-        data={tableData}
-        shouldFixTableLayout={!isMobile}
-        minRows={tableData.length}
-        dense={false}
-        showHeader={!isMobile}
-      />
+      <div className={clsx(isDesktop ? styles.searchAndFilterDesktop : styles.searchAndFilter)}>
+        <PureSearchBarWithBackdrop
+          value={searchString}
+          onChange={(e: any) => setSearchString(e.target.value)}
+          isSearchActive={!!searchString}
+          placeholderText={placeHolderText}
+          zIndexBase={zIndexBase}
+          isDesktop={isDesktop}
+          className={clsx(isDesktop ? styles.searchBarDesktop : styles.searchBar)}
+        />
+        <div className={clsx(isDesktop ? styles.searchResultsDesktop : styles.searchResults)}>
+          {searchResultsText}
+        </div>
+      </div>
+      {hasSearchResults ? (
+        <Table
+          kind={TableKind.V2}
+          alternatingRowColor={true}
+          columns={animalsColumns}
+          data={filteredInventory}
+          shouldFixTableLayout={isDesktop}
+          minRows={filteredInventory.length}
+          dense={false}
+          showHeader={isDesktop}
+        />
+      ) : (
+        <NoSearchResults
+          className={clsx(isDesktop ? styles.noSearchResultsDesktop : styles.noSearchResults)}
+          searchTerm={searchString}
+          includeFiltersInClearSuggestion
+        />
+      )}
```

Comment:
I looked into "+ Load more" button being shown sometimes.
In `Table`, `minRows` is the initial value of `data.length` and won't change.

In `Inventory` component, if you have 10 animals and filter out 6 animals and refresh the page, `minRows` is `4` and when you clear filters, "Load more" button shows up.

The same thing happens after `hasSearchResults` is changed to `false` to `true` here since `Table` is re-rendered.

(I hope it makes sense)

I will make a change to pass `inventory.length` as `minRows` from the container in my PR later.

Reply:
Nice! Right that makes sense I think it doesn't need to be dynamic unless data is refetched.